app/SocialMedia/Base.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * Abstract Social media class.
+ *
+ * @package   App
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+
+namespace App\SocialMedia;
+
+/**
+ * Abstract Social media class.
+ */
+abstract class Base
+{
+	/**
+	 * Allowed type of log.
+	 */
+	public const ALLOWED_TYPE_OF_LOG = ['info', 'warning', 'error'];
+
+	/**
+	 * @var string
+	 */
+	protected static $socialMediaType;
+
+	/**
+	 * Base constructor.
+	 *
+	 * @param string $userName
+	 */
+	abstract public function __construct(string $userName);
+
+	/**
+	 * Is configured.
+	 *
+	 * @return bool
+	 */
+	abstract public static function isActive();
+
+	/**
+	 * Remove mass social media records from DB.
+	 *
+	 * @param string[] $logins
+	 *
+	 * @return mixed
+	 */
+	abstract public static function removeMass(array $logins);
+
+	/**
+	 * Retrieve data from Api.
+	 */
+	abstract public function retrieveDataFromApi();
+
+	/**
+	 * Remove social media account from database.
+	 */
+	abstract public function remove();
+
+	/**
+	 * Log info.
+	 *
+	 * @param string $message
+	 *
+	 * @throws \App\Exceptions\AppException
+	 * @throws \yii\db\Exception
+	 */
+	public static function logInfo(string $message)
+	{
+		static::log('info', $message);
+	}
+
+	/**
+	 * Log error.
+	 *
+	 * @param string $message
+	 *
+	 * @throws \App\Exceptions\AppException
+	 * @throws \yii\db\Exception
+	 */
+	public static function logError(string $message)
+	{
+		static::log('error', $message);
+	}
+
+	/**
+	 * Log warning.
+	 *
+	 * @param string $message
+	 *
+	 * @throws \App\Exceptions\AppException
+	 * @throws \yii\db\Exception
+	 */
+	public static function logWarning(string $message)
+	{
+		static::log('warning', $message);
+	}
+
+	/**
+	 * Log to db.
+	 *
+	 * @param string $typeOfLog
+	 * @param string $message
+	 *
+	 * @throws \App\Exceptions\AppException
+	 * @throws \yii\db\Exception
+	 */
+	public static function log(string $typeOfLog, string $message)
```

Comment:
typeOfLog > type

Reply:
full logs to App\Log::log() and general information for synchronization to l_#__social_media_logs